### DISCUSSION TITLE
Instansiating a class once per CRD

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/dotnet-operator-sdk#548] [Q&A]
Hi there,

Is it possible for me to use a class that instantiated only once per CRD? This class behaves as a monitor/supervisor for the app running _inside_ my pods. The app itself is clustered (meaning the pods are joined together into this app's cluster), and I need to be able to monitor and react to state changes within it. I do not need to know the state of the Kubernetes cluster from within the app.

This class needs to:
- Instantiate when a CRD is applied and stay alive until that CRD is deleted (or the operator is shut down). This means I need one instance of the class per CRD
- Be able to use interfaces like ILogger<type> and the KubeOps IKubernetesClient 

Is it possible to do this? How might I implement it?

Thanks for this awesome SDK by the way. It has saved us a lot of development work. 😎 

Regards,

Ian

**Comment by buehler** [comment]
Hey @MasterofTofu 
You're welcome :-)

If I understand you correctly, I'd say this is either possible or a misconception 😁 
If you have registered your CRD within Kubernetes and you create an instance of the CRD, then an object in the operator is created. This object (class) is not changed as long as the operator runs. However, it is updated if a change to the CRD instance is made.

If you just want to monitor stuff, you can use the operator like normal and just watch for your CRD. When the reconcile loop runs, you can use your logger and whatnot and just hold a reference to the UID of the instance.

So basically, what you describe is normally the way you use an operator (if I understand correctly)
